src/devices/Hmc5883l/Gain.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Iot.Device.Hmc5883l
+{
+    /// <summary>
+    /// HMC5883L Gain Setting
+    /// </summary>
+    public enum Gain
+    {
+        /// <summary>
+        /// 1370, recommended sensor field range: ±0.88 Ga
+        /// </summary>
+        Gain1 = 0x00,
```

Comment:
It is a bit strange the way we have it here, since when using it you have to do `Gain.Gain1` which doesn't look very nice. Are these names (Gain1, Gain2, etc) the same as the datasheet? if not, then we can try something different like `Range.Gain1` or `Gain.1370` or something like that.